test/test.js

```diff
@@ -0,0 +1,259 @@
+/*
+* @Author: alutun
+* @Date:   2017-07-03 14:01:38
+* @Last Modified by:   alutun
+* @Last Modified time: 2017-07-03 16:19:16
+*/
+
+var supertest = require("supertest");
+var should = require("should");
+var dateFormat = require('dateformat');
+var now = new Date();
+var util = require('util');
+
+var server = supertest.agent("http://localhost:3000");
+
+var savedId;
+// UNIT test begin
+
+describe("Basic test",function(){
+  it("should return home page",function(done){
+    // Test home page API
+    server
+    .get("/")
+    .expect("Content-type",/json/)
+    .expect(200) // THis is HTTP response
+    .end(function(err,res){
+      // HTTP status should be 200
+      res.status.should.equal(200);
+      done();
+    });
+  });
+
+  it("Should return 404 for wrong endpoint", function(done){
+  	server
+  	.get("/random")
+    .expect(404)
+    .end(function(err,res){
+      res.status.should.equal(404);
+      done();
+  	});
+   });
+});
+
+describe("PUT /user",function(){
+
+	it("Simple Put Test", function(done){
+		server
+		.post("/user")
+	 	.send({name: "toto", dob: dateFormat(now, "isoDateTime"), address: "Shanghai", description: "This user is a test"})
+		.expect("Content-type",/json/)
+	    .expect(200)
+	    .end(function(err,res){
+	      res.status.should.equal(200);
+	      // console.log("res body is : " + util.inspect(res.body));
+	      res.body.name.should.equal("toto");
+	      res.body.description.should.equal("This user is a test");
+	      res.body.address.should.equal("Shanghai");
+	      res.body.should.have.property("id").which.is.a.Number();
+	      savedId = res.body.id;
```

Comment:
Hello @xavierchow let me rephrase me your comment to make sure I understand your point before I submit a commit .

You mean that because `saveId` is defined in the global scope, it's an issue because that link the different batch of test together because they rely on each other result. Is that right ?